compiler/src/iree/compiler/Codegen/Utils/Utils.h

```diff
@@ -275,9 +275,35 @@ void sinkOpsInCFG(const SmallVector<Operation *> &allocs,
 // the inputs.
 bool hasFusedLeadingOp(linalg::LinalgOp rootOp);
 
+// Utility to make sure we are storing the full incoming subspan. Otherwise we
+// cannot simply adjust the subspan's resultant type later.
+bool isFullSlice(OffsetSizeAndStrideOpInterface sliceLoadStoreOp,
+                 IREE::TensorExt::DispatchTensorType tensorType,
+                 ValueRange dynamicDims);
+
+//===---------------------------------------------------------------------===//
+// Scalable size utility functions
+//===---------------------------------------------------------------------===//
+
 std::optional<vector::VscaleRange>
 getDefaultVscaleRange(IREE::HAL::ExecutableTargetAttr targetAttr);
 
+using SizesAndScalableFlags =
+    std::pair<SmallVector<int64_t>, SmallVector<bool>>;
+
+// This utility function returns the static part of scalable inner tile sizes.
+// These are - as of now always and probably should always be - arith.muli ops
+// with `vector.vscale` on one side and an `arith.constant` on the other. It
+// returns the constant if found.
+FailureOr<int64_t> getStaticPartOfScalableTileSize(Operation *op);
```

Comment:
Should this be a local function given the only user is `getScalableTileSizesAndFlags`?